vochain/scrutinizer/vote.go

```diff
@@ -254,15 +254,24 @@ func (s *Scrutinizer) ComputeResult(processID []byte) error {
 	if err != nil {
 		return fmt.Errorf("computeResult: cannot load processID %x from database: %w", processID, err)
 	}
-
 	// Compute the results
 	var results *indexertypes.Results
 	if results, err = s.computeFinalResults(p); err != nil {
 		return err
 	}
-	p.HaveResults = true
-	p.FinalResults = true
-	if err := s.db.Update(processID, p); err != nil {
+
```

Comment:
Can you add a comment here exlaining the RACE and why we need to use updateMatching? So future developments not get into the same issue